### PR TITLE
Allow filtering documents with tag synonyms

### PR DESCRIPTION
**Pull request by fugal-dy**
This adds the TagSynonymGroup model and extends the
documents tagfilter such that filtering for document
by tag looks for associated synonyms to include.

Issue #133

### CODE REVIEW

**Review by winged — CHANGES_REQUESTED** [review]
Good work so far!

There are a couple of things missing:

* Register the new viewset under **alexandria/core/urls.py**
* Your `black` was a bit overzealous - check the version we use vs. your version

File: alexandria/core/models.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 from django.utils.translation import gettext_lazy as _
 from localized_fields.fields import LocalizedCharField, LocalizedTextField
 from rest_framework import exceptions
+from django.conf import settings
```
[the remark this answers is not in the document]
Comment (fugal-dy) [reply alexandria/core/models.py:11]:
```suggestion
```
True. Incomplete code cleanup after removing localization that was using `settings.LANGUAGES` for `choices`  and `settings.LANGUAGE_CODE` for `default`

**Comment by fugal-dy** [comment]
I have applied your inputs, fixed the filter (did not do what it should) and did some cleanup. On top I added an extra test that does not make it past linting though o_0. I need some sleep now, will get back to finishing this later

**Review by winged — CHANGES_REQUESTED** [review]

**Comment by Yelinz** [comment]
Seems to have been merged with https://github.com/projectcaluma/alexandria/pull/256 already, reopen if not

**Pull request closed without merging**